### DISCUSSION TITLE
Licence GPL -> MIT

### DISCUSSION THREAD

**Discussion by melashkov** [ain/smartbanner.js#643] [Q&A] [answered]
Hey, very cool library, would you consider changing your license to MIT?

**Comment by ain** [comment, marked as answer]
Thanks for the question.

GPL-3 licence was intentionally chosen to promote Free Software.

The main goal was to make sure that changes would be more likely to return to the community and that it'd not be ripped off into closed-source commercial distributions without benefitting the community that has spent numerous hours of unpaid work to make it happen.

On these grounds I hope you understand I have to deny your suggestion for MIT-licence.

**Comment by Timvde** [comment]
I just came across this issue. I support GPL and its ideas, so I understand why you don't like the MIT license. But since GPL is so viral, I wonder if it makes more sense to use [LGPL](https://en.wikipedia.org/wiki/GNU_Lesser_General_Public_License), which is more common for libraries. With GPL, I believe that everyone using your library in their website would need to make their entire front-end GPL. I don't think that's what you're after. LGPL would protect the code of this library, without infecting unrelated code.

**Comment by ain** [comment]
Thanks for the heads up.

There's nothing that'd stop one to use GPL-licenced components on a website. The important aspect here is that this component would remain free to its users regardless of who makes a change to it or distributes it.

LGPL would allow the component to be bundled into and distributed within the closed silo which I referred to in the previous comment.

**Comment by Timvde** [comment]
LGPL would not allow for any modifications to your code being closed.

I do think that you can't legally use a GPL library on an otherwise closed-source website. Client-side Javascript is "distributed software" (and is therefore not entailed by the loophole that AGPL closes). Because of the viral nature of GPL, this means that the entire front-end (or at least the entire JS stack) needs to be GPLed. If that's what you want, then you should of course keep it GPL. If not (i.e. you want to allow people to use your library without opening up their entire website front-end, as long as they contribute back changes to your library), I believe LGPL is a better fit.

**Comment by manufunk** [comment]
Sorry that I have to come back to this topic. After some legal discussions that I had in my company, we are sadly not allowed to use this library. As we are also supporting open source software and would love to contribute, we can only do so with a license that does not force us to reveal all the related source code. Therefore again the question if something like LGPL would be possible.

**Comment by ddurham2** [comment]
I also have concerns with GPL3.  How can one possibly use this in a non-open web frontends (e.g. those that are minified, typescript->js compiled, dart->js compiled, etc) without having to make the original source public?  TMK it can't.  It didn't sound like that was your original intent, but that is (I believe) the unintended effect.

@ain, I humbly submit that the [AGPL](https://en.wikipedia.org/wiki/GNU_Affero_General_Public_License) would be a better license for what you stated in the second comment:
> It is intended for software designed to be run over a [network](https://en.wikipedia.org/wiki/Computer_network), adding a provision requiring that the corresponding [source code](https://en.wikipedia.org/wiki/Source_code) of modified versions of the software be prominently offered to all users who interact with the software over a network.

**Comment by ddurham2** [comment]
Any thoughts on whether linking (hosted elsewhere) to the .js, in the <head>, unmodified, as it is distributed by the author, triggers any requirements of the GPL3 to release the rest of the web code?
